datasets/template_dataset.py

```diff
@@ -0,0 +1,159 @@
+from benchopt import BaseDataset
+from benchopt import safe_import_context
+
+# Protect the import with `safe_import_context()`. This allows:
+# - skipping import to speed up autocompletion in CLI.
+# - getting requirements info when all dependencies are not installed.
+with safe_import_context() as import_ctx:
+    import numpy as np
+    from libsvmdata import fetch_libsvm
+
+    import jax
+    import jax.numpy as jnp
+    from functools import partial
+
+    from jaxopt import LBFGS
+
+
+def loss_sample(inner_var, outer_var, x, y):
+    return -jax.nn.log_sigmoid(y*jnp.dot(inner_var, x))
+
+
+def loss(inner_var, outer_var, X, y):
+    batched_loss = jax.vmap(loss_sample, in_axes=(None, None, 0, 0))
+    return jnp.mean(batched_loss(inner_var, outer_var, X, y), axis=0)
+
+
+# All datasets must be named `Dataset` and inherit from `BaseDataset`
+class Dataset(BaseDataset):
+    """Hyperparameter optimization with IJCNN1 dataset."""
+    # Name to select the dataset in the CLI and to display the results.
+    name = "ijcnn1"
+    """How to add a new problem to the benchmark?
+
+    This template dataset is an adaptation of the dataset from the benchopt
+    template benchmark (https://github.com/benchopt/template_benchmark/) to
+    the bilevel setting.
+    """
+
+    install_cmd = 'conda'
+    # List of packages needed to run the dataset. See the corresponding
+    # section in objective.py
+    requirements = ['pip:libsvmdata', 'scikit-learn']
+
+    # List of parameters to generate the datasets. The benchmark will consider
+    # the cross product for each key in the dictionary.
+    # Any parameters 'param' defined here is available as `self.param`.
+    parameters = {
+        'reg_parametrization': ['exp'],
+    }
+
+    def get_data(self):
+        # The return arguments of this function are passed as keyword arguments
+        # to `Objective.set_data`. This defines the benchmark's
+        # API to pass data.
```

Comment:
This docstring should specify the output of the function; what do you need to define in this function?